src/vm/jitinterface.cpp

```diff
@@ -11113,7 +11113,7 @@ void CEEJitInfo::CompressDebugInfo()
     } CONTRACTL_END;
 
     // Don't track JIT info for DynamicMethods.
-    if (m_pMethodBeingCompiled->IsDynamicMethod())
+    if (m_pMethodBeingCompiled->IsDynamicMethod() && CLRConfig::GetConfigValue(CLRConfig::EXTERNAL_TrackDynamicMethodDebugInfo) == 0)
```

Comment:
This should use the same EEConfig setting I suggested caching above. GetConfigValue [isn't exactly fast](https://github.com/dotnet/coreclr/blob/a28b25aacdcd2adb0fdfa70bd869f53ba6565976/src/utilcode/clrconfig.cpp#L201).